File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        boolean isA = true;
        if(s.length() != t.length()){
            isA = false;
        }else{
            Map<Character, Integer> count = new HashMap<>();
            for(int i = 0; i < s.length(); ++i){
                count.put(s.charAt(i), count.getOrDefault(s.charAt(i), 0) + 1);
            }
            for(int i = 0; i < t.length() && isA; ++i){
                count.put(t.charAt(i), count.getOrDefault(t.charAt(i), 0) - 1);
                if(count.get(t.charAt(i)) == -1){
                    isA = false;
                }
            }
        }
        return isA;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            d.update({s[i] : d.get(s[i], 0) + 1})
            d.update({t[i] : d.get(t[i], 0) - 1})
            if d.get(s[i]) == 0:
                d.pop(s[i])
            if d.get(t[i]) == 0:
                d.pop(t[i])
        return d == {}